docs/user-guide/replicasets.md

```diff
@@ -0,0 +1,91 @@
+Controller only supports equality-based selector requirements.
+
+Most [`kubectl`](kubectl/kubectl.md) commands that support Replication Controllers also
+support Replica Sets. Commands that support Replica Sets are:
+[`scale`](kubectl/kubectl_scale.md), [`autoscale`](kubectl/kubectl_autoscale.md),
+[`expose`](kubectl/kubectl_expose.md). One exception is the
+[`rolling-update`](kubectl/kubectl_rolling-update.md) command. If you want the rolling
+update functionality please consider using Deployments instead. Also, the
+[`rolling-update`](kubectl/kubectl_rolling-update.md) command is imperative
+whereas Deployments are declarative, so we recommend using Deployments through the
+[`rollout`](kubectl/kubectl_rollout.md) command.
+
+While replica sets can be used independently, today it's mainly used by
+[Deployments](deployments.md) as a mechanism to orchestrate pod creation, deletion and
+updates. When you use Deployments you don't have to worry about manging the replica sets
+that they create. Deployments own and manage their replica sets.
+
+### Example
+
+{% include code.html language="yaml" file="replicasets/frontend.yaml" ghlink="/docs/user-guide/replicasets/frontend.yaml" %}
+
+Saving this config into `frontend.yaml` and submitting it to a Kubernetes cluster should
+create the defined replica set and the pods that it manages.
+
+```shell
+$ kubectl create -f frontend.yaml 
+replicaset "frontend" created
+$ kubectl describe rs/frontend
+Name:		frontend
+Namespace:	default
+Image(s):	gcr.io/google_samples/gb-frontend:v3
+Selector:	tier=frontend,tier in (frontend)
+Labels:		app=guestbook,tier=frontend
+Replicas:	3 current / 3 desired
+Pods Status:	3 Running / 0 Waiting / 0 Succeeded / 0 Failed
+No volumes.
+Events:
+  FirstSeen	LastSeen	Count	From				SubobjectPath	Type		Reason			Message
+  ---------	--------	-----	----				-------------	--------	------			-------
+  1m		1m		1	{replicaset-controller }			Normal		SuccessfulCreate	Created pod: frontend-qhloh
+  1m		1m		1	{replicaset-controller }			Normal		SuccessfulCreate	Created pod: frontend-dnjpy
+  1m		1m		1	{replicaset-controller }			Normal		SuccessfulCreate	Created pod: frontend-9si5l
+$ kubectl get pods
+NAME             READY     STATUS    RESTARTS   AGE
+frontend-9si5l   1/1       Running   0          1m
+frontend-dnjpy   1/1       Running   0          1m
+frontend-qhloh   1/1       Running   0          1m
+```
+
+## Replica Set as an Horizontal Pod Autoscaler target
+
+A Replica Set can also be a target for [Horizontal Pod Autoscalers (HPA)](horizontal-pod-autoscaler.md).
+Here is an example HPA targeting the Replica Set we created in the previous example.
+
+```yaml
+apiVersion: autoscaling/v1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: frontend-scaler
+spec:
+  scaleRef:
+    kind: ReplicaSet
+    name: frontend
+    subresource: scale
+  minReplicas: 3
+  maxReplicas: 10
+  cpuUtilization:
+    targetPercentage: 50
+```
```

Comment:
We can `kubectl create xxx` this HPA to auto-scale the RS. 

Alternatively, we can also use `kubectl autoscale xxx` ...